### PR TITLE
Addition (3rd_party_board): Add board support for Autosport Labs ESP32-CAN-X2 development board

### PR DESCRIPTION
**Pull request by brentpicasso**
----------
## Description of Change
Adds support for Autosport Labs ESP32-CAN-X2 development board

## Tests scenarios
Tested with the development board

## Related links
https://wiki.autosportlabs.com/ESP32-CAN-X2

### CODE REVIEW

File: variants/aslcanx2/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define EXTERNAL_NUM_INTERRUPTS 46
+#define NUM_DIGITAL_PINS        48
+#define NUM_ANALOG_INPUTS       20
```
Comment (P-R-O-C-H-Y) [inline variants/aslcanx2/pins_arduino.h:12]: Those defines were moved from variants and are defined globally per SoC. Please remove it to fix warnings.
```suggestion
```

File: variants/aslcanx2/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+static const uint8_t LED_BUILTIN = 2;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+#define LED_BUILTIN LED_BUILTIN
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<48)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 46)
```
Comment (P-R-O-C-H-Y) [inline variants/aslcanx2/pins_arduino.h:20]: Same here. Those defines were moved from variants and are defined globally per SoC. Please remove it to fix warnings.
```suggestion
```

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@brentpicasso Please take a look on my comments.

File: boards.txt [hunk 3]
```diff
@@ -36447,3 +36447,205 @@ weact_studio_esp32c3.menu.EraseFlash.all=Enabled
...
+aslcanx2.serial.disableDTR=false
+aslcanx2.serial.disableRTS=false
+
+aslcanx2.build.tarch=xtensa
+aslcanx2.build.bootloader_addr=0x0
+aslcanx2.build.target=esp32s3
+aslcanx2.build.mcu=esp32s3
+aslcanx2.build.core=esp32
+aslcanx2.build.variant=aslcanx2
+aslcanx2.build.board=ESP32S3_DEV
```
Comment (P-R-O-C-H-Y) [inline boards.txt:36478]: I recommend to change the build.board to be your board specific name, for example ASL_CAN_X2.

File: variants/aslcanx2/pins_arduino.h (hunk 1 above)
Comment (brentpicasso) [reply variants/aslcanx2/pins_arduino.h]: thanks, I manually applied this change.

File: variants/aslcanx2/pins_arduino.h (hunk 2 above)
Comment (brentpicasso) [reply variants/aslcanx2/pins_arduino.h]: thanks, I manually applied this change.

File: boards.txt (hunk 3 above)
Comment (brentpicasso) [reply boards.txt:36478]: Thanks, I changed it to ASL-CAN-X2.

Comment (P-R-O-C-H-Y) [reply boards.txt:36478]: Please use underscores instead of dashes. Thanks!

Comment (brentpicasso) [reply boards.txt:36478]: Done! :) thank you.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by brentpicasso** [comment]
@P-R-O-C-H-Y  thank you for the quick approval.  Basic question - once this is merged, does it automatically show up in a nightly build of the Arduino IDE? Or is there a way existing IDE installations can get this update?  Googling didn't yield much info. Thanks again!

**Comment by P-R-O-C-H-Y** [comment]
> @P-R-O-C-H-Y thank you for the quick approval. Basic question - once this is merged, does it automatically show up in a nightly build of the Arduino IDE? Or is there a way existing IDE installations can get this update? Googling didn't yield much info. Thanks again!

You are welcome :) The nightly builds of Arduino IDE are unrelated to the ESP32 Arduino core.
Once we merge it there will be 2 ways to get the board to the Arduino IDE:
- Wait until next release 3.0.1 of the ESP32 Arduino core (this repo) and download it through the Boards manager in the IDE.
- Use manual installation of the ESP32 Arduino core (master branch) and get the boards immediately after it's merged to master. You can find the documentation of manual installation [HERE](https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#windows-manual-installation), search for your OS :)

**Comment by lucasssvaz** [comment]
@brentpicasso For some reason the pre-commit bot is unable to push to your branch, maybe it could be related to organization permissions if your repo is in one. 

To fix that we can try enabling "Allow edits by maintainers" in this PR or alternatively, as it is a small PR, you can manually apply the changes shown [here](https://github.com/espressif/arduino-esp32/actions/runs/9270474998/job/25515801880?pr=9683)

**Comment by brentpicasso** [comment]
> @brentpicasso For some reason the pre-commit bot is unable to push to your branch, maybe it could be related to organization permissions if your repo is in one.
> 
> To fix that we can try enabling "Allow edits by maintainers" in this PR or alternatively, as it is a small PR, you can manually apply the changes shown [here](https://github.com/espressif/arduino-esp32/actions/runs/9270474998/job/25515801880?pr=9683)

Thanks, manually push the requested fix, including a correction to match product's PID / VID.

File: variants/aslcanx2/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#define Pins_Arduino_h
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x16D0
+#define USB_PID 0x07F2
+
+static const uint8_t LED_BUILTIN = 2;
+#define BUILTIN_LED LED_BUILTIN // backward compatibility
```
Comment (lucasssvaz) [inline variants/aslcanx2/pins_arduino.h:11]:
```suggestion
#define BUILTIN_LED LED_BUILTIN  // backward compatibility
```

File: variants/aslcanx2/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+static const uint8_t CAN1_RX = 6;
+static const uint8_t CAN1_TX = 7;
+
+static const uint8_t SS = -1;
+static const uint8_t CS = 10;
+static const uint8_t MOSI = 11;
+static const uint8_t MISO = 13;
+static const uint8_t SCK = 12;
+
```
Comment (lucasssvaz) [inline variants/aslcanx2/pins_arduino.h:22]:
```suggestion
```

**Review by lucasssvaz — CHANGES_REQUESTED** [review]
Just for the CI to pass

**Review by lucasssvaz — APPROVED** [review]

**Pull request merged**